framework/quota/src/org/apache/cloudstack/quota/vo/ServiceOfferingVO.java

```diff
@@ -310,14 +269,12 @@ public String getDeploymentPlanner() {
     }
 
     public String getDetail(String name) {
-        assert (details != null) : "Did you forget to load the details?";
-
-        return details != null ? details.get(name) : null;
+        if (details == null) throw new IllegalStateException("Cannot set value as details is null");
+        return details.get(name);
     }
 
     public void setDetail(String name, String value) {
-        assert (details != null) : "Did you forget to load the details?";
-
+        if (details == null) throw new IllegalStateException("Cannot set value as details is null");
```

Comment:
I think that would work (returning an empty map) though we need to check the history as to why we never did that earlier. For now, the fix LGTM as it's better than the assert. We can take the hashmap fix separately.